caikit/core/data_model/base.py

```diff
@@ -37,6 +37,10 @@
 from ..exceptions import error_handler
 from . import enums, json_dict, timestamp
 
+if TYPE_CHECKING:
+    # Local
+    from caikit.core.data_model.data_backends import DataModelBackendBase
+    from caikit.interfaces.common.data_model.file import File
```

Comment:
This is what is currently causing CI to fail `tox -e imports`/`./scripts/check_deps.sh`.

I'm assuming `pydeps` has no concept of `TYPE_CHECKING` resulting in this test failing even though this is only being imported when running in type checking mode. I can comment this out, but it would be interesting to have working type checking

Reply:
Yeah, that makes sense. I think this may have slipped through my review, but looking more closely, since `File` is _itself_ a `DataBase` derived class, I think we need to rethink this a bit. It's used as a forward declaration here for the `to_file` method, but I think we may want to change that method to instead return something more python-native.

Thoughts @joerunde?

Reply:
I ran a few tests with `mypy` and right now I'm getting `Found 761 errors in 84 files (checked 156 source files)`. So we can leave this commented out and get back to it when/if we decide to tackle typing.

Reply:
I ended up commenting this out and adding pragmas to ignore warnings for now.